app/models/activity.rb

```diff
@@ -222,8 +222,14 @@ def enroll_reservists!
     # Filter non-masters if masters-only, non-freshmen if freshman-only.
     # Note: this will leave nobody if someone enables both is_masters and
     # is_freshmans, as freshman? explicitly rejects masters.
-    reservistpool.select! { |m| m.member.masters? } if is_masters?
-    reservistpool.select! { |m| m.member.freshman? } if is_freshmans?
+    if is_freshmans? || is_penultimates? || is_seniors? || is_masters?
```

Comment:
Dislike this method, I'm sure there is a better way

Reply:
Probably a good idea to factor out this logic and the logic above and put it together in a single method